src/videojs-hls.js

```diff
@@ -995,7 +1019,7 @@ videojs.HlsHandler.prototype.loadSegment = function(segmentInfo) {
       return;
     }
 
-    self.lastSegmentLoaded_ = segmentInfo.uri;
+    self.lastSegmentLoaded_ = segment;
```

Comment:
I don't think this works. Elsewhere in this file we're comparing `lastSegmentLoaded` to a URL which will always fail after this change.